### PR TITLE
Update jenkins CI from ROCm 5.2 to ROCm 5.3

### PR DESCRIPTION
**Pull request by Rombur**
We cannot build Docker images based on ROCm 5.2 anymore because AMD removed an external repository need to build the image. We need to bump the version from 5.2 to 5.3. ROCm 5.3 is also the oldest version on Frontier.

### CODE REVIEW

**Review by dalg24 — APPROVED** [review]
AFAICT we still have coverage with Sandia's nightlies
Do we need to update the testing information on the minimum requirements page in the doc?

**Review by JBludau — APPROVED** [review]

**Comment by Rombur** [comment]
> Do we need to update the testing information on the minimum requirements page in the doc?

Yes, I'll do it

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
```
[  9%] Building CXX object core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_HIP.dir/UnitTestMainInit.cpp.o
/var/jenkins/workspace/Kokkos_PR-7588/core/unit_test/hip/TestHIP_InterOp_Graph.cpp:107:42: error: use of undeclared identifier 'hipGraphDebugDotFlagsVerbose'
      graph.native_graph(), dot.c_str(), hipGraphDebugDotFlagsVerbose));
                                         ^
[  9%] Building CXX object core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_OpenMPInterOp.dir/openmp/TestOpenMP_InterOp.cpp.o
[  9%] Building CXX object core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_HIP.dir/hip/TestHIP_Abort.cpp.o
[ 10%] Linking CXX executable Kokkos_CoreUnitTest_CTestDevice
1 error generated when compiling for gfx90a.
make[2]: *** [core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_HIPInterOpGraph.dir/build.make:79: core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_HIPInterOpGraph.dir/hip/TestHIP_InterOp_Graph.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1792: core/unit_test/CMakeFiles/Kokkos_CoreUnitTest_HIPInterOpGraph.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

**Comment by Rombur** [comment]
Fixed. We need ROCm 5.5 to write a DOT file.

**Comment by dalg24** [comment]
> Fixed. We need ROCm 5.5 to write a DOT file.

How comes were did not hit hat before, was it because the underlying standard library implementation was too old and we were skipping the test?
Is there a link to the rocm doc you could share?

**Comment by Rombur** [comment]
> How comes were did not hit hat before, was it because the underlying standard library implementation was too old and we were skipping the test?

We cannot use the native graph in ROCm 5.2 because of a bug. This test is only for native implementations

https://github.com/kokkos/kokkos/blob/26c1c33c07df7b75152d34fce8d104a023982747/core/unit_test/hip/TestHIP_InterOp_Graph.cpp#L39-L42

> Is there a link to the rocm doc you could share?

You need to search for `hipGraphDebugDotPrint` in the [release note](https://rocm.docs.amd.com/en/docs-5.5.0/release.html)

**Review by dalg24 — DISMISSED** [review]
You need to skip the test, not just comment out what does not compile, so that we don't get the false impression it ran

**Review by romintomasetti — DISMISSED** [review]
Are we changing the minimum requirement to `5.3` ? If so, we should get rid of all the `KOKKOS_IMPL_HIP_NATIVE_GRAPH` related things.

**Comment by dalg24** [comment]
> Are we changing the minimum requirement to `5.3` ? If so, we should get rid of all the `KOKKOS_IMPL_HIP_NATIVE_GRAPH` related things.

We're not changing the minimum requirement at this time.

File: core/unit_test/hip/TestHIP_InterOp_Graph.cpp [hunk 1]
```diff
@@ -101,6 +101,9 @@ TEST(TEST_CATEGORY, graph_instantiate_and_debug_dot_print) {
 
   ASSERT_EQ(num_nodes, 2u);
 
+  // hipGraphDebugDotPrint was introduced in ROCm 5.5
+#if (HIP_VERSION_MAJOR > 5) || \
+    ((HIP_VERSION_MAJOR > 5) && (HIP_VERSION_MINOR >= 5))
```
Comment (romintomasetti) [inline core/unit_test/hip/TestHIP_InterOp_Graph.cpp:106]: I tend to agree with @dalg24 that the whole test could be skipped in this case. Not blocking though.

**Comment by Rombur** [comment]
We still check that we built a graph with the correct number of nodes. We just don't check the name of the kernels in the graph. I agree that the test is not as strong but it is still useful and I don't think we should skip it.

**Review by romintomasetti — APPROVED** [review]
